custom_completions.go

```diff
@@ -284,7 +298,7 @@ func (c *Command) getCompletions(args []string) (*Command, []string, ShellCompDi
 
 	var completions []string
 	directive := ShellCompDirectiveDefault
-	if flag == nil {
+	if flag == nil && flagCompletion {
```

Comment:
I'm unsure about this one.  I believe it disables subcommand completion and `ValidArgs` completion for the non-interspersed case, right?

I think `ValidArgs` should still be done.
And I'm thinking subcommand completion should also be done. case like:
```
testprog --bool subcomm<TAB>
```
What do you think?

Reply:
You are correct this is wrong. It works fine without it. I added a test for it.